heralding/capabilities/rdp.py

```diff
@@ -68,122 +54,102 @@ class RDP(HandlerBase):
             session.end_session()
 
     async def _handle_session(self, reader, writer, session):
-        data = await reader.read(2048)
-        cr_pdu = x224ConnectionRequestPDU()
-        cr_pdu.parse(data)
-
-        start_tls = True  # this forces tls security
-        client_reqProto = 1  # set default to tls
-        if cr_pdu.reqProtocols:
-            client_reqProto = cr_pdu.reqProtocols
-        else:
-            # if no nego request was made, then it is rdp security
-            client_reqProto = 0
-
-        cc_pdu_obj = x224ConnectionConfirmPDU(client_reqProto, start_tls)
-        cc_pdu = cc_pdu_obj.getFullPacket()
-        writer.write(cc_pdu)
-        await writer.drain()
-        if cc_pdu_obj.sentNegoFail:
-            logger.debug("Sent x224 RDP Negotiation Failure PDU")
-            session.end_session()
-            return
-        logger.debug("Sent x244CLinetConnectionConfirm PDU")
+        try:
+            data = await reader.read(2048)
```

Comment:
Why don’t you use recv_next_tpkt here?

Reply:
earlier, i did that because it is non-tls packet. but now it should change